amethyst_gltf/src/format/importer.rs

```diff
@@ -0,0 +1,320 @@
+use std;
+use std::error::Error as StdError;
+use std::fmt;
+use std::path::Path;
+use std::sync::Arc;
+
+use assets::BoxedErr;
+use assets::Source as AssetSource;
+use base64;
+use gltf;
+use gltf::Gltf;
+use gltf::json;
+use gltf::json::validation;
+use gltf_utils::Source;
+
+#[derive(Debug)]
+pub enum ImageFormat {
+    Png,
+    Jpeg,
+}
+
+impl ImageFormat {
+    fn from_mime_type(mime: &str) -> Self {
+        match mime {
+            "image/jpeg" => ImageFormat::Jpeg,
+            "image/png" => ImageFormat::Png,
+            _ => unreachable!(),
+        }
+    }
+}
+
+/// Buffer data returned from `import`.
+#[derive(Clone, Debug)]
+pub struct Buffers(Vec<Vec<u8>>);
+
+impl Source for Buffers {
+    fn source_buffer(&self, buffer: &gltf::Buffer) -> &[u8] {
+        &self.0[buffer.index()]
+    }
+}
+
+#[allow(unused)]
+impl Buffers {
+    /// Obtain the contents of a loaded buffer.
+    pub fn buffer(&self, buffer: &gltf::Buffer) -> Option<&[u8]> {
+        self.0.get(buffer.index()).map(Vec::as_slice)
+    }
+
+    /// Obtain the contents of a loaded buffer view.
+    pub fn view(&self, view: &gltf::buffer::View) -> Option<&[u8]> {
+        self.buffer(&view.buffer()).map(|data| {
+            let begin = view.offset();
+            let end = begin + view.length();
+            &data[begin..end]
+        })
+    }
+
+    /// Take the loaded buffer data.
+    pub fn take(self) -> Vec<Vec<u8>> {
+        self.0
+    }
+}
+
+fn import_impl(source: Arc<AssetSource>, path: &Path) -> Result<(Gltf, Buffers), Error> {
+    let data = read_to_end(source.clone(), path)?;
+    if data.starts_with(b"glTF") {
+        import_binary(&data, source, path)
+    } else {
+        import_standard(&data, source, path)
+    }
+}
+
+/// Imports glTF 2.0
+pub fn import<P>(source: Arc<AssetSource>, path: P) -> Result<(Gltf, Buffers), Error>
+where
+    P: AsRef<Path>,
+{
+    import_impl(source, path.as_ref())
```

Comment:
You could replace this with the body of `import_impl` and just add

```rust
let path = path.as_ref()
```

at the top.